### ISSUE TITLE
auto beautifying php file sometimes make the file all gone

### ISSUE THREAD

**Issue by Linghui** [Glavin001/atom-beautify#481]
run beautify directly is ok for php file.
but if turn 'Beautify On Save' on PHP and try to save many tims, sometimes it make the file empty.
I didn't try other types of file, so I am not sure if the same would happen to others.

**Comment by Glavin001** [comment]
> but if turn 'Beautify On Save' on PHP and try to save many tims, sometimes it make the file empty.

Yeah.. I am not surprised. There is a lot of room for improvement in this respect: making sure corruptions do not occur.

Atom's Text Editors and Atom Beautify have a very special relationship and sometimes when you beautify a file quickly like this, it ends up beautifying an empty text editor. To elaborate, Atom Beautify's PHP beautification is asynchronous and so making the changes to your file after completing the beautification, it will be in a state of flux, where Atom Beautify is trying to write to the file and maybe the file is temporarily empty. If you re-run beautify when it's in this nondeterministic state it can load the file as if it were empty, beautify it (beautify the empty file), and then overwrite the file with an empty beautification result (because the input file it thought was empty). 

Long story short: try not to save and/or beautify so quickly in succession. Beautify On Save is currently the worst offender, because it has to work around Atom in some cases and everything is async and without conflict resolution for speedy-cases like this.

Please let me know if any other bugs occur. Thanks.

**Comment by KillerJK** [comment]
Just wanted to say I'm also seeing this problem with scss files (csscomb) when beautify on save is being used. Thanks for the explanation, Glavin.

**Comment by johnsto** [comment]
I've seen this happen on pretty much kind of file I use regularly (JS, LESS, and HTML). Had to disable the "beautify on save" option and switched to manual operation until a solution can be found.

**Comment by Glavin001** [comment]
Unfortunately, I do not have much time after school and work and spending time with family to debug this issue and publish a fix. Maybe over the holidays in December I can find time. If someone finds a solution and can submit a Pull Request. I will happily review and merge it. Thank you!
